### ISSUE TITLE
rollback removal of alternc-ssl from 3.x

### ISSUE THREAD

**Issue by lelutin** [AlternC/AlternC#416]
To avoid problems when upgrading from 3.3.x to 3.5.0, we need to bring back alternc-ssl and make a change to it so that it does not remove all SSL vhosts.

The change is already adressed by #374. This issue is only about performing the commit rollback to bring back the alternc-ssl package, and it should happen before #374 can move forward.

**Comment by kienanstewart** [comment]
Pre-upgrade w/ the patch AlternC 3.3.12 (from the 416-fix_alterc-ssl_purge branch):

```
# mysql alternc -e 'select * from sub_domaines';
+----+--------+-------------+-----+-------------------------+--------------+------------+------------+---------+
| id | compte | domaine     | sub | valeur                  | type         | web_action | web_result | enable  |
+----+--------+-------------+-----+-------------------------+--------------+------------+------------+---------+
|  3 |   2000 | example.com |     | https://www.example.com | url-ssl      | OK         | 0          | ENABLED |
|  5 |   2000 | example.com | mix | /www/example.com/       | vhost-mixssl | OK         | 0          | ENABLED |
|  6 |   2000 | example.com | www | /www/example.com/       | vhost-ssl    | OK         | 0          | ENABLED |
+----+--------+-------------+-----+-------------------------+--------------+------------+------------+---------+
```

Then test upgrade to 3.5rc:

```
root@debian-9-amd64:/home/vagrant# apt install /vagrant/alternc_3.5.0~rc1_all.deb
Suggested packages:
  alternc-awstats alternc-mailman alternc-changepass amavisd-new clamav-daemon spamassassin pop-before-smtp magpierss ncompress alternc-squirrelmail alternc-roundcube
Recommended packages:
  ntp alternc-certificate-provider-letsencrypt
The following packages will be REMOVED:
  alternc-ssl
The following packages will be upgraded:
  alternc
1 upgraded, 0 newly installed, 1 to remove and 39 not upgraded.
Need to get 0 B/1,077 kB of archives.
After this operation, 109 kB of additional disk space will be used.

root@debian-9-amd64:/home/vagrant# alternc.install
Running upgrade script 3.5.0.1.sql
ERROR 1062 (23000) at line 69: Duplicate entry 'fqdn_dovecot' for key 'name'
ERROR 1062 (23000) at line 73: Duplicate entry 'fqdn_postfix' for key 'name'
ERROR 1062 (23000) at line 77: Duplicate entry 'fqdn_proftpd' for key 'name'
Running upgrade script 3.5.0.2.php
Running upgrade script 3.5.0.2.sql
ERROR 1064 (42000) at line 3: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'UPDATE alternc_status SET value='3.5.0.2.sql' WHERE name='alternc_version'' at line 5
Running upgrade script 3.5.0.3.sql
ERROR 1091 (42000) at line 4: Can't DROP 'compte'; check that column/key exists
ERROR 1071 (42000) at line 5: Specified key was too long; max key length is 767 bytes
Expanding variables in configuration files: etc/alternc/bureau.conf etc/apache2/envvars etc/alternc/apache2.conf etc/alternc/phpmyadmin.inc.php etc/bind/named.conf.options etc/postfix/master.cf etc/postfix/myalias.cf etc/postfix/myrelay.cf etc/postfix/mydomain.cf etc/postfix/myrelay-domain.cf etc/postfix/mymail2mail.cf etc/postfix/mygid.cf etc/postfix/myquota.cf etc/postfix/myvirtual.cf etc/postfix/mytransport.cf etc/postfix/sasl/smtpd.conf etc/alternc/postfix/postfix.cf etc/alternc/postfix/postfix-slave.cf etc/opendkim.conf etc/default/opendkim etc/proftpd/proftpd.conf etc/proftpd/welcome.msg etc/proftpd/modules.conf etc/default/saslauthd etc/dovecot/alternc-sql.conf etc/dovecot/alternc-dict-quota.conf etc/dovecot/conf.d/95_alternc.conf etc/dovecot/conf.d/96_ssl.conf.
Adding user `dovecot' to group `ssl-cert' ...
Adding user dovecot to group ssl-cert
Done.
Adding user `postfix' to group `ssl-cert' ...
Adding user postfix to group ssl-cert
Done.
Considering dependency mpm_prefork for mpm_itk:
Considering conflict mpm_event for mpm_prefork:
Considering conflict mpm_worker for mpm_prefork:
Module mpm_prefork already enabled
Module mpm_itk already enabled
Enabling module expires.
To activate the new configuration, you need to run:
  systemctl restart apache2
Enabling conf alternc-ssl.
To activate the new configuration, you need to run:
  systemctl reload apache2
Policy already active, not updated
Fix all the permission. May be quite long...
YOU CAN INTERUPT THIS BY USING Ctrl-c THEN y TO BYPASS THE ERROR.
AlternC fixperms.sh -> working on /var/www/alternc/a/admin                                                                                                                                                         
Done
Compile PO files
This script will rebuild all web configuration and regenerate DNS.
Use --force to skip confirmation
++ Start rebuilding ++
Set flag to rebuild
Now launching update_domains to rebuild.
Finish.

root@debian-9-amd64:/home/vagrant# mysql alternc -e 'select * from sub_domaines';
+----+--------+-------------+-----+-------------------------+-------+------------+------------+---------+----------------+----------+-------+
| id | compte | domaine     | sub | valeur                  | type  | web_action | web_result | enable  | certificate_id | provider | https |
+----+--------+-------------+-----+-------------------------+-------+------------+------------+---------+----------------+----------+-------+
|  3 |   2000 | example.com |     | https://www.example.com | url   | OK         | 0          | ENABLED |              0 |          |       |
|  5 |   2000 | example.com | mix | /www/example.com/       | vhost | OK         | 0          | ENABLED |              0 |          | both  |
|  6 |   2000 | example.com | www | /www/example.com/       | vhost | OK         | 0          | ENABLED |              0 |          | https |
+----+--------+-------------+-----+-------------------------+-------+------------+------------+---------+----------------+----------+-------+
```

Looks good now

**Comment by kienanstewart** [comment]
To limit the scope of the work, I've set up the branch 416-fix_alterc-ssl_purge to have the wheezy and jessie patches applied, plus cherry-picked the changes from #222 so that it can be installed on stretch. It should also install and run on jessie, but I haven't tested that exhaustively - it requires accepting that the key for the repos is expired. I don't think it's unreasonable to target just jessie/stretch with 3.3.12

**Comment by kienanstewart** [comment]
PR #436

**Comment by camlafit** [comment]
Normally is no more required as 3.5 and plugins should/must manage correctly all use case.

If we encounter some bug during update process at least 3.5rc2, please to open a new issue :)
